File: workspaces-good/modules/shared/src/taxonomy_result_vo.py
from typing import Optional

from .taxonomy_operation_vo import OperationVO


class ResultVO:
    def __init__(self, expression: str, value: float):
        self.expression = expression
        self.value = value

    def __repr__(self) -> str:
        return self.expression


def create_result(
    left: float, op: OperationVO, right: float, value: float
) -> ResultVO:
    return ResultVO(
        expression=f"{left} {op.value} {right} = {value}",
        value=value,
    )

File: workspaces-good/packages/shared/src/taxonomy_result_vo.ts
export interface ResultVO {
  expression: string;
  value: number;
}

export function createResult(
  left: number,
  op: string,
  right: number,
  value: number,
): ResultVO {
  return {
    expression: `${left} ${op} ${right} = ${value}`,
    value,
  };
}
